### ISSUE TITLE
livereload is triggered even when `livereloadFilterPatterns` matches changed file (when multiple files change)

### ISSUE THREAD

**Issue by bantic** [ember-cli/ember-cli#6191]
The `liveReloadFilterPatterns` project option allows one to specify an array of regex patterns that, if any match the file that initiated a rebuild, will prevent ember-cli from notifying livereload of the changed files. This happens in the [`shouldTriggerReload` method of tasks/livereload-server.js](https://github.com/ember-cli/ember-cli/blob/93d2994817dc46cddda9eaf952ce48c6047f5054/lib/tasks/server/livereload-server.js#L127).

However, the watcher that the livereload server is using (broccoli-sane-watcher) only reports the *first changed file* if multiple files changed within the debounce time to trigger the rebuild. (See https://github.com/broccolijs/broccoli-sane-watcher/issues/35)

Because of this, if two (or more) files change at nearly the same time, the first one "wins" and the livereload trigger is only skipped if that first file matches any of the `liveReloadFilterPatterns`. For example, if the following line is added to `ember-cli-build.js`:
```javascript
// ember-cli-build.js
// ...
app.project.liveReloadFilterPatterns = [ /application\.hbs/ ];
```

Then ember-cli should avoid triggering a livereload if application.hbs was changed. That correctly happens: if I `touch app/template/application.hbs`, I see this console output from ember-cli:
```
file changed templates/application.hbs

Build successful - 281ms.
Skipping livereload for: app/templates/application.hbs
```

But if I touch two files at roughly the same time, the first one wins and it influences whether ember-cli triggers the rebuild. For instance if I `touch app/template/application.hbs && touch app/styles/app.css` then I see this output:
```
file changed templates/application.hbs
file changed styles/app.css

Build successful - 277ms.
Skipping livereload for: app/templates/application.hbs
```

But if I reverse the order of the touch command to: `touch app/styles/app.css && touch app/templates/application.hbs` then the `shouldTriggerReload` only sees the filePath "app/styles/app.css" (which doesn't match any of the live reload filter patterns) and reload does happen.

I've made the suggestion at broccoli-sane-watcher that it could report all files that were changed. If that makes sense then this quasi-bug in ember-cli could be fixed by having `shouldTriggerReload` consider all the changed files and avoiding rebuild if any of them match the patterns.

For context, in the PR in which "hot CSS reloading" landed there is some discussion (that seems relevant to this issue) about whether using the file(s) that changed to trigger the build or the files that changed as a result of the build should be used to decide whether to do the livereload: https://github.com/ember-cli/ember-cli/pull/5165#issuecomment-160506062.

---

Output from ember version --verbose:
```
ember-cli: 2.7.0
http_parser: 2.6.0
node: 5.4.1
v8: 4.6.85.31
uv: 1.8.0
zlib: 1.2.8
ares: 1.10.1-DEV
icu: 56.1
modules: 47
openssl: 1.0.2e
os: darwin x64
```

**Comment by stefanpenner** [comment]
yup seems funky:
https://github.com/ember-cli/ember-cli/blob/master/lib/tasks/server/livereload-server.js#L157 likely pants to pass in `files` as well, so here
https://github.com/ember-cli/ember-cli/blob/master/lib/tasks/server/livereload-server.js#L127-L129 we can do the correct thing.

**Comment by bantic** [comment]
Yeah, except the `filePath` is the path to the file before ember-cli builds it, and `files` is the list of resulting changed files after the build finishes.
So for example if the `app.css` has an `@import "other.css"` line and someone modifies `other.css`, then `filePath` would be the path to `other.css` and `files` would include `app.css` (but probably not `other.css` since it probably isn't in the build output).
So they are sort of apples and oranges. But I also think that the post-build `files` should be considered for the live reload trigger either way. I will roll some thoughts about this into the RFC I'll write for https://github.com/ember-cli/ember-cli/issues/6192#issuecomment-240747752.

**Comment by mikemunsie** [comment]
I dived into the code and noticed it was using an `every` (https://github.com/ember-cli/ember-cli/blob/master/lib/tasks/server/livereload-server.js#L126) comparison through the array of patterns. I believe the expected behavior is that if at least one pattern is matched, it should return true. I changed to this to use `some` instead and my filters are now working as expected. 

PR @ https://github.com/ember-cli/ember-cli/pull/7265
